c++/src/kj/async.h

```diff
@@ -370,6 +370,10 @@ class Promise: protected _::PromiseBase {
   // Returns a dump of debug info about this promise.  Not for production use.  Requires RTTI.
   // This method does NOT consume the promise as other methods do.
 
+  void ignore() {}
```

Comment:
I think this is too easy to mix up with `ignoreResult()`. Can we tell people to use a `(void)` cast instead?

Alternatively, we could call this `cancel()`, and have it actually do `node = nullptr;`. That's possibly a nice API to have, although not exactly the semantics you intended here.

Reply:
Definitely have to agree that `ignore()` would be confusing with `ignoreResult()` and would probably make it easier to introduce bugs unfortunately. I like the `(void)` cast approach here in general as it is a clear indicator that something "special" is happening.

Reply:
Done. ignore() is no more.

Reply:
The benefit of `ignore()` is that you can easily look up what it means. Not so with `(void)`. Could we introduce a different name that won't be confused with other operations and be easier to look up its docs? Maybe `drop` or `discard` or even `discardPromise`?

Reply:
Not my business, but `(void)var;` works with any type / return value annotated with `[[nodiscard]]`.
That is a general way of telling the compiler to shut up.
As a KJ user, knowing this rule, I prefer to use it than to have special case in my head for this kj type.

Reply:
@dom96 A `(void)` cast to ignore a discard warning is a widely-used idiom in C and C++; it's not something invented here. It's the right thing to do because it's what most people have seen many times before.

Anyway you almost never want to ignore this warning regardless so this won't come up much.

Reply:
FWIW I googled "cast to void" and got this as the first result: https://stackoverflow.com/questions/34288844/what-does-casting-to-void-really-do